### PR TITLE
[basevotes / funvotes] Ignore vote actions if target is no longer available

### LINKED ISSUE #641: [CS:GO] Basevotes error
```
L 07/27/2017 - 22:39:23: [SM] Exception reported: Client index 5 is invalid
L 07/27/2017 - 22:39:23: [SM] Blaming: basevotes.smx
L 07/27/2017 - 22:39:23: [SM] Call stack trace:
L 07/27/2017 - 22:39:23: [SM]   [0] BanClient
L 07/27/2017 - 22:39:23: [SM]   [1] Line 349, /home/builds/sourcemod/linux-1.8/build/plugins/basevotes.sp::Handler_VoteCallback
```

### PR DESCRIPTION
**Pull request by Timocop**
Checks the target userid at the end of the vote and ignores the actions if userid is invalid instead of throwing exceptions.
Fixes #641.

### CODE REVIEW

File: plugins/funvotes/voteburn.sp [hunk 1]
```diff
@@ -40,6 +40,8 @@ void DisplayVoteBurnMenu(int client, int target, char[] name)
 	}
 	
 	g_voteClient[VOTE_CLIENTID] = target;
```
Comment (asherkin) [inline plugins/funvotes/voteburn.sp]: `VOTE_CLIENTID` should probably go away completely.

File: plugins/funvotes.sp [hunk 2]
```diff
@@ -247,19 +247,35 @@ public int Handler_VoteCallback(Menu menu, MenuAction action, int param1, int pa
...
 				case (slay):
 				{
-					PrintToChatAll("[SM] %t", "Slayed player", g_voteInfo[VOTE_NAME]);					
-					LogAction(-1, g_voteClient[VOTE_CLIENTID], "Vote slay successful, slaying \"%L\"", g_voteClient[VOTE_CLIENTID]);
-					
-					ExtinguishEntity(g_voteClient[VOTE_CLIENTID]);
-					ForcePlayerSuicide(g_voteClient[VOTE_CLIENTID]);
+					if(GetClientOfUserId(g_voteClient[VOTE_USERID]) == 0)
+					{
+						PrintToChatAll("[SM] %t", "Player no longer available");
```
Comment (asherkin) [inline plugins/funvotes.sp]: I don't think we need prints on failure (the logging can stay) - people know the client has gone.

One day these should all be changed to `ShowActivity`, but not in this PR.

Comment (Timocop) [inline plugins/funvotes.sp]: Isnt it better to let the clients know the vote has actually ended?
People may know the client has gone, but they dont know if the vote actually ended unless they start a new vote.

**Comment by Timocop** [comment]
Everything ok with this?

**Comment by asherkin** [comment]
I think so, and I've dismissed my reject, but I'd like someone else to take a pass over it before merging.

**Comment by ErikMinekus** [comment]
Instead of putting the player name in the vote failed messages, I would use ```%L``` with ```voteTarget```, to be consistent with the vote successful messages.

And instead of literally putting "Player no longer available" as the reason for failing, I would use ```%T``` and pass ```LANG_SERVER``` as an additional argument, so that the English phrase is used.

If you want me to nitpick about the code style:
```sourcepawn
int voteTarget;
if((voteTarget = GetClientOfUserId(g_voteTarget)) == 0)
```
I would write this as:
```sourcepawn
int voteTarget = GetClientOfUserId(g_voteTarget);
if (voteTarget == 0)
```
And instead of using else and having another indentation level, you could ```return 0``` after the vote fails.

**Comment by Timocop** [comment]
Logging should always be in english, or do you want to translate logs on AlliedModders all the time if someone has a problem? You can look in other AlliedModders plugins, they do the same. Same as the code style you mentioned.

**Comment by ErikMinekus** [comment]
I forgot that the server language can be changed in the configs, in that case the code is fine.

**Comment by Headline22** [comment]
Looks good to me 👍

**Comment by asherkin** [comment]
Thanks for this!

**Pull request merged**